File: cis_interface/examples/timed_pipe/src/timed_pipe_src.m
function timed_pipe_src(msg_count, msg_size)

  msg_count = str2num(msg_count);
  msg_size = str2num(msg_size);
  fprintf('Hello from Matlab pipe_src: msg_count = %d, msg_size = %d\n', ...
          msg_count, msg_size);

  % Ins/outs matching with the the model yaml
  outq = YggInterface('YggOutput', 'output_pipe');
  disp('pipe_src(M): Created I/O channels');

  % Send test message multiple times
  test_msg(1:msg_size) = '0';
  count = 0;
  for i = 1:msg_count
      ret = outq.send(test_msg);
      if (~ret)
          error(sprintf('pipe_src(M): SEND ERROR ON MSG %d\n', i));
      end;
      count = count + 1;
  end;

  fprintf('Goodbye from Matlab source. Sent %d messages.\n', count);

end

File: cis_interface/examples/timed_pipe/src/timed_pipe_src.cpp
#include <stdio.h>
#include "YggInterface.hpp"

int main(int argc, char *argv[]) {
  if (argc != 3) {
    printf("Error in C++ pipe_src: The message count and size must be provided as input arguments.\n");
    return -1;
  }
  int exit_code = 0;
  int ret = 0;

  int msg_count = atoi(argv[1]);
  int msg_size = atoi(argv[2]);
  printf("Hello from C++ pipe_src: msg_count = %d, msg_size = %d\n",
	 msg_count, msg_size);

  // Ins/outs matching with the the model yaml
  YggOutput outq("output_pipe");
  printf("pipe_src(CPP): Created I/O channels\n");

  // Create test message
  char *test_msg = (char*)malloc(msg_size + 1);
  int i;
  for (i = 0; i < msg_size; i++)
    test_msg[i] = '0';
  test_msg[i] = '\0';

  // Send test message multiple times
  int count = 0;
  for (i = 0; i < msg_count; i++) {
    ret = outq.send(test_msg, msg_size);
    if (ret < 0) {
      printf("pipe_src(CPP): SEND ERROR ON MSG %d\n", i);
      exit_code = -1;
      break;
    }
    count++;
  }

  printf("Goodbye from C++ source. Sent %d messages.\n", count);
  free(test_msg);
  return exit_code;
}

